### PR TITLE
Add dual-screen support to Calculator

### LINKED ISSUE #1029: Calculator should support dual-portrait and dual-landscape on 10X
**Problem Statement**
The current graphing calculator UI isn't optimize for Windows 10X dual screens.

**Proposal**
Modify the layout to take advantage of dual-portrait and dual-landscape postures

Inkedpostures-overview_LI

**Requested Assignment**
If possible, I would like to implement this.

### PR DESCRIPTION
**Pull request by rudyhuyn**
## Fixes #1029

Support Dual-landscape and Dual-portrait modes.

image
image
image
image

### Description of the changes:
- Add new StateTrigger detecting the different view modes
- Add a fork of TwoPaneView in C++CX for 2 reasons:
   - The fix for C++ apps (crash when navigating) isn't available for the moment
   - Add `Pane1MinLength`, `Pane1MaxLength`, `Pane2MinLength`, `Pane2MaxLength` properties
- Merge Visual State Group of Unit Converter, causing some issues due to conflicts
- Add `DeviceFamilyHelper`

### Other changes
Visual states of `Calculator.xaml` had some collisions, the states related to AOT (https://github.com/microsoft/calculator/blob/master/src/Calculator/Views/Calculator.xaml#L521) and the states related to CalculationResult (https://github.com/microsoft/calculator/blob/master/src/Calculator/Views/Calculator.xaml#L544) were both triggered by the same type of trigger (window resizing), without checking if the app was AOT or not, and were modifying the same properties which caused some layouts issues.

In order to solve this problem, which also affects dual screens, we have replaced the StateTriggers used in AOT visual states with `ApplicationViewModeTrigger` in order to activate them only when the application is actually AOT.

### How changes were validated:
- manually with 10X 10.0.19563

### CODE REVIEW

File: src/Calculator/Calculator.vcxproj.filters [hunk 1]
```diff
@@ -328,6 +328,9 @@
       <Filter>Common</Filter>
     </ClCompile>
     <ClCompile Include="Common\ViewState.cpp" />
+    <ClCompile Include="ApplicationViewModeTrigger.cpp">
+      <Filter>Views\StateTriggers</Filter>
+    </ClCompile>
```
Comment (EriWong) [inline src/Calculator/Calculator.vcxproj.filters]: We should probably move ApplicationViewModeTrigger.cpp to the views folder and have the file structure line up with the project structure.

**Comment by rudyhuyn** [comment]
@joseartrivera I've extended the scope to this PR and fixes some issues with visual states in Calculator.xaml (see description). Ready for review!

File: src/Calculator/Views/StateTriggers/ApplicationViewModeTrigger.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+{
+    auto applicationView = ApplicationView::GetForCurrentView();
+    auto viewMode = applicationView->ViewMode;
+    if (applicationView->VisibleBounds.Width < this->MinWindowWidth || applicationView->VisibleBounds.Height < this->MinWindowHeight)
+    {
+        SetActive(false);
+        return;
+    }
+
+    switch ((int)viewMode)
```
Comment (joseartrivera) [inline src/Calculator/Views/StateTriggers/ApplicationViewModeTrigger.cpp]:
```suggestion
    switch (static_cast<int>(viewMode))
```

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml [hunk 3]
```diff
@@ -383,26 +386,60 @@
...
+                        <Setter Target="PaneView.TallModeConfiguration">
+                            <Setter.Value>
+                                <muxc:TwoPaneViewTallModeConfiguration>TopBottom</muxc:TwoPaneViewTallModeConfiguration>
+                            </Setter.Value>
+                        </Setter>
+                    </VisualState.Setters>
+                </VisualState>
+                <VisualState>
+                    <VisualState.StateTriggers>
+                        <states:ApplicationViewModeTrigger ViewMode="DoublePortrait"/>
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml]: Can we make KeyGraphFeaturesControl span 2 columns in double portrait?
image

Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml]: Weird layout when rotating in double portrait
image

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml [hunk 4]
```diff
@@ -411,246 +448,257 @@
...
+                                    <CornerRadius x:Key="RightButtonCornerRadius">0</CornerRadius>
+                                    <Style x:Key="ThemedGrapherStyle" TargetType="graphControl:Grapher">
+                                        <Setter Property="AxesColor" Value="{ThemeResource SystemColorWindowTextColor}"/>
+                                        <Setter Property="GraphBackground" Value="{ThemeResource SystemColorWindowColor}"/>
+                                    </Style>
+                                </ResourceDictionary>
+                            </ResourceDictionary.ThemeDictionaries>
+                        </ResourceDictionary>
+                    </Grid.Resources>
+                    <graphControl:Grapher Name="GraphingControl"
```
Comment (joseartrivera) [inline src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml]: Not sure if related to this PR, but when on a single pane the graph surface is zoomed in and offset on the default view
image

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml (hunk 3 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml]: Yes, it happens when you span the window (double-landscape) and then rotate (double-portrait), I will investigate

Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml]: same issue than above, probably the state trigger not in sync with the twopaneview

File: src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml (hunk 4 above)
Comment (rudyhuyn) [reply src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml]: Yes, this is an issue with the GraphControl we will need to investigate (other PR)

**Pull request merged**